### PR TITLE
Update OIDC library & Swiss Featureset

### PR DESCRIPTION
**Pull request by omarhoblos**

### CODE REVIEW

File: .github/workflows/docker-test.yml [hunk 1]
```diff
@@ -8,13 +8,24 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - name: checkout code
-        uses: actions/checkout@v2
-      - name: install buildx
+      - name: Checkout
```
Comment (AD1306) [inline .github/workflows/docker-test.yml:11]: is this file even needed, I don't see any usages ?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -49,12 +41,30 @@ const routes: Routes = [
   imports: [
     BrowserModule,
     HttpClientModule,
-    OAuthModule.forRoot(),
     FormsModule,
     ReactiveFormsModule,
     FontAwesomeModule,
     AppRoutingModule,
-    RouterModule.forRoot(routes),
+    AuthModule.forRoot({
```
Comment (AD1306) [inline src/app/app.module.ts:48]: i believe you should move this to a module of its own and return an array of configs. It'll be less of a breaking change later on https://nice-hill-002425310.azurestaticapps.net/docs/documentation/configuration#using-multiple-configs

File: src/app/components/fhirdata/fhirdata.component.ts [hunk 3]
```diff
@@ -145,37 +149,39 @@ export class FhirdataComponent implements OnInit {
...
-      this.showLoadingBar = true;
-      this.utilService.resetErrorObject(this.errorObject);
-      this.httpService.getFhirQueries(query, headers).subscribe(
-        recordsFound => {
-          const result = this.setupDataInView(recordsFound);
-          if (result === 'dataFound') {
-            this.bundleFound = recordsFound;
+    this.oidcSecurityService.isAuthenticated().subscribe(isAuthenticated => {
+      if (isAuthenticated) {
+        this.showLoadingBar = true;
```
Comment (AD1306) [inline src/app/components/fhirdata/fhirdata.component.ts:165]: move this statements inside if conditions to a method of its own for better readability.

File: src/app/components/home/home.component.html [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
+  <ng-container *ngIf="returnTokenStatus()">
     <div>
       <h2>Your Tokens</h2>
       <div class="card">
-        <span id="expiryHeader">
-          Expires on: {{tokenObjectForDisplay['expirationDate']}}
-        </span>
+        <ng-container *ngIf="expirationMinutes">
+          <span id="expiryHeader">
+            Expires in {{expirationMinutes}} minutes, on {{tokenObjectForDisplay['expirationDateForDisplay']}}
```
Comment (AD1306) [inline src/app/components/home/home.component.html:23]: love this !!

File: src/app/components/home/home.component.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    });
+  }
+
+  ngAfterViewInit(): void {
+    if (this.expirationMinutes >= 0) {
+      this.checkExpiration()
+    }
+  }
+
+  returnTokenStatus() {
```
Comment (AD1306) [inline src/app/components/home/home.component.ts:60]: i believe this is not used.

File: fhirserverinstructions/federatedscript.md [hunk 6]
```diff
@@ -45,6 +45,10 @@ Add the following script to your application's OIDC Server Definition
     theOutcome.addAuthority('FHIR_READ_ALL_OF_TYPE', 'Location');
     theOutcome.addAuthority('FHIR_READ_ALL_OF_TYPE', 'Practitioner');
 
+    if (theContext.getStringClaim('identifier').length > 0) {
+        theOutcome.setUserData('federated_detail', theContext.getStringClaim('identifier'));
+    }
```
Comment (AD1306) [inline fhirserverinstructions/federatedscript.md:50]:
```suggestion
    const identifier = theContext.getStringClaim('identifier');
    if (identifier.length > 0) {
        theOutcome.setUserData('federated_detail', identifier);
    }
```

what do you think of this ?

File: src/app/components/home/home.component.ts (hunk 5 above)
Comment (omarhoblos) [reply src/app/components/home/home.component.ts:60]: This is still being used in the template for some hide/ show logic

File: src/app/app.module.ts (hunk 2 above)
Comment (omarhoblos) [reply src/app/app.module.ts]: Migrated to its own separate auth module

File: src/app/components/fhirdata/fhirdata.component.ts (hunk 3 above)
Comment (omarhoblos) [reply src/app/components/fhirdata/fhirdata.component.ts:164]: Migrated to a separate private function in the component

File: .github/workflows/docker-test.yml (hunk 1 above)
Comment (omarhoblos) [reply .github/workflows/docker-test.yml:11]: Yup, this is being used for the Docker CI test

**Review by AD1306 — APPROVED** [review]

**Pull request merged**